joomla/joomla-cms
# Fix install languages and discover extension for mssql and small fixes for other databases

Pull Request for Issue #9457 and related to #11763 which is incomplete.

### Summary of Changes

Mysql strict mode - install languages fix.
Postgresql - reinstall languages fix, prevent double installation in db.
Mssql - install and reinstall and discover extension fix.

__Additional changes:__
- column xreference across all databases has default value ''
- asset_id on mysql has to be unsigned.

Mysql column type TEXT can not have default value ''. It is ignored.
Source:  https://dev.mysql.com/doc/refman/5.7/en/blob.html
> BLOB and TEXT columns cannot have DEFAULT values. 

So from others db I also removed default value for:
- data (#__updates)
- custom_data (#__extensions)
- system_data (#__extensions)

### Testing Instructions
[MYSQL - strict mode]
1. Install Joomla on mysql and enable strict mode, for ex change one line:
```diff
diff --git a/libraries/joomla/database/driver/mysqli.php b/libraries/joomla/database/driver/mysqli.php
index 37c9ae0f44..9c9e10206d 100644
--- a/libraries/joomla/database/driver/mysqli.php
+++ b/libraries/joomla/database/driver/mysqli.php
@@ -184,7 +184,8 @@ class JDatabaseDriverMysqli extends JDatabaseDriver
                }
 
                // Set sql_mode to non_strict mode
-               mysqli_query($this->connection, "SET @@SESSION.sql_mode = '';");
+               mysqli_query($this->connection, "SET @@SESSION.sql_mode = 'ONLY_FULL_GROUP_BY,STRICT_TRANS_TABLES,ERROR_FOR_DIVISION_BY_ZERO,NO_AUTO_CREATE_USER,NO_ENGINE_SUBSTITUTION';");
```
2. Try to install languages, it should failed
3. apply patch
4. Now should be OK.

[POSTGRESQL]
1. Install joomla on postgresql.
2. Install some language pack, reinstall it. before patch language will be installed twice.
3. After patch it is installed only once

[SQLSRV]
1. Install joomla on sqlsrv
2. Try to install some language - it fails
3. Try to discover some extension - it fails (you should have one not installed, but extracted in correct place)
4. Apply patch
5. Install language
6. Try to discover extension

### Expected result
All above things works

### Documentation Changes Required
None
